### PR TITLE
Fix OVSDB/Netlink races due to incomplete ovsdb interface entry

### PR DESCRIPTION
**Pull request by safchain**
skip compile-tests
skip functional-tests-backend-orientdb
skip functional-hw-tests
skip scale-tests
skip cdd-overview-tests
skip k8s-tests
skip unit-tests
skip go-fmt

### CODE REVIEW

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by safchain** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
@safchain I took the liberty the change your fix a bit : instead of removing the previous netlink node, I changed the way the OVS probe looks for an existing node. If there is no previous MAC address, just use the name and the ifindex (https://github.com/skydive-project/skydive/pull/2265/commits/6dc6f88c99050aa958ebfcb15b77badc81d42b60#diff-581c1aab9b2e6b5dbcb0698934ee8f43R331-R338). It also passed CI 5-6 times. If it looks ok to you, can we merge this ?

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run functional-tests-backend-elasticsearch

File: topology/probes/ovsdb/ovsdb.go [hunk 1]
```diff
@@ -325,16 +326,22 @@ func (o *Probe) OnOvsInterfaceAdd(monitor *ovsdb.OvsMonitor, uuid string, row *l
...
 	intf := o.Ctx.Graph.LookupFirstNode(graph.Metadata{"UUID": uuid})
-	if mac != "" || attachedMAC != "" {
-		var macFilter *filters.Filter
-		if mac != "" {
-			macFilter = filters.NewTermStringFilter("MAC", mac)
-		} else {
-			macFilter = filters.NewTermStringFilter("MAC", attachedMAC)
+	if oldMAC != "" || mac != "" || attachedMAC != "" {
+		var macFilters []*filters.Filter
+		if oldMAC != "" {
```
Comment (safchain) [inline topology/probes/ovsdb/ovsdb.go:331]: any reason to not use the mac or attachedMAC if the previous MAC was empty ?

**Comment by safchain** [comment]
I just left a comment, otherwise looks good, nice to see this fixed :)

File: topology/probes/ovsdb/ovsdb.go (hunk 1 above)
Comment (lebauce) [reply topology/probes/ovsdb/ovsdb.go:331]: The behaviour I observed was:
- The interface is first seen by netlink but with a strange MAC address
- The OVS probe then handles the update. It has a valid MAC address but no previous MAC. So we don't have any other way to find the existing interface (neither with `mac`nor `oldMAC`).

Comment (safchain) [reply topology/probes/ovsdb/ovsdb.go:331]: I don't see why we need to ensure that we have a previous MAC ? If the netlink one is incorrect and the OVS is correct then they are different and the `lookup` will fail.

Anyway let merge this version which fixes the CI and I'll try to submit another patch to understand this behavior.

Comment (lebauce) [reply topology/probes/ovsdb/ovsdb.go:331]: Agreed :-)

**Pull request merged**